Unity 2022.1.10 MessagePack 2.4.35 IL2CPP build Issues

Hi there. 
I can't build IL2CPP project for PC. Trying NET standart and NET Framework
Editor serialization works fine.

[Google Drive Link ](https://drive.google.com/drive/folders/1q8PNXeSw_l1FLfFLeI5Y9RlI5kwNvxIi?usp=sharing)to repro project and full Editor log

Part of Editor log message:

Building Library\Bee\artifacts\WinPlayerBuildProgram\u1oik\lhyy_ssagePack2.obj failed with output:
MessagePack2.cpp
D:\Projects\Temp\TestMP\Library\Bee\artifacts\WinPlayerBuildProgram\il2cppOutput\cpp\MessagePack2.cpp(20747): error C2664: 'String_t *Strings_WeakIntern_mC72EF2A3C712D84A4E92074BC83E711414D7F0BD(ReadOnlySpan_1_t0846CD1BEF355AE12F9A41FFB292AA4A85F072F3,const RuntimeMethod *)': cannot convert argument 1 from 'ReadOnlySpan_1_t59614EA6E51A945A32B02AB17FBCBDF9A5C419C1' to 'ReadOnlySpan_1_t0846CD1BEF355AE12F9A41FFB292AA4A85F072F3'
D:\Projects\Temp\TestMP\Library\Bee\artifacts\WinPlayerBuildProgram\il2cppOutput\cpp\MessagePack2.cpp(20747): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
D:\Projects\Temp\TestMP\Library\Bee\artifacts\WinPlayerBuildProgram\il2cppOutput\cpp\MessagePack2.cpp(7047): note: see declaration of 'Strings_WeakIntern_mC72EF2A3C712D84A4E92074BC83E711414D7F0BD'
UnityEditor.GenericMenu:CatchMenu (object,string[],int)

--- a/src/MessagePack.UnityClient/build.bat
+++ b/src/MessagePack.UnityClient/build.bat
@@ -1,7 +1,7 @@
 @echo off
 setlocal
 
-IF NOT EXIST "%~dp0Assets/Microsoft.VisualStudio.Threading.dll" CALL "%~dp0copy_assets.bat"
+IF NOT EXIST "%~dp0Assets/Plugins/System.Runtime.CompilerServices.Unsafe.dll" CALL "%~dp0copy_assets.bat"
 
 IF "%Build_ArtifactStagingDirectory%"=="" (
     SET LogDirectory=%~dp0..\..\bin\build_logs

--- a/src/MessagePack.UnityClient/build.sh
+++ b/src/MessagePack.UnityClient/build.sh
@@ -8,7 +8,7 @@ done
 
 SCRIPT_DIR=$(dirname "$(realpath $0)")
 
-if ! [[ -e "${SCRIPT_DIR}/Assets/Plugins/System.Memory.dll" ]] ; then
+if ! [[ -e "${SCRIPT_DIR}/Assets/Plugins/System.Runtime.CompilerServices.Unsafe.dll" ]] ; then
    ${SCRIPT_DIR}/copy_assets.sh
 fi
 

--- a/src/MessagePack.UnityClient/copy_assets.bat
+++ b/src/MessagePack.UnityClient/copy_assets.bat
@@ -8,10 +8,7 @@
 
 @pushd %~dp0
 
-echo F | xcopy "..\..\bin\MessagePack\release\netstandard2.0\publish\System.Buffers.dll" ".\Assets\Plugins\System.Buffers.dll" /Y /I
-echo F | xcopy "..\..\bin\MessagePack\release\netstandard2.0\publish\System.Memory.dll" ".\Assets\Plugins\System.Memory.dll" /Y /I
 echo F | xcopy "..\..\bin\MessagePack\release\netstandard2.0\publish\System.Runtime.CompilerServices.Unsafe.dll" ".\Assets\Plugins\System.Runtime.CompilerServices.Unsafe.dll" /Y /I
 echo F | xcopy "..\..\bin\MessagePack\release\netstandard2.0\publish\Microsoft.NET.StringTools.dll" ".\Assets\Plugins\Microsoft.NET.StringTools.dll" /Y /I
-echo F | xcopy "..\..\bin\MessagePack\release\netstandard2.0\publish\System.Threading.Tasks.Extensions.dll" ".\Assets\Plugins\System.Threading.Tasks.Extensions.dll" /Y /I
 
 @popd

--- a/src/MessagePack.UnityClient/copy_assets.sh
+++ b/src/MessagePack.UnityClient/copy_assets.sh
@@ -18,8 +18,5 @@ if ! [[ -d "${SCRIPT_DIR}/Assets/Plugins" ]] ; then
   mkdir -p ${SCRIPT_DIR}/Assets/Plugins
 fi
 
-cp ${SCRIPT_DIR}/../../bin/MessagePack/${BUILDCONFIGURATION}/netstandard2.0/publish/System.Buffers.dll ${SCRIPT_DIR}/Assets/Plugins/System.Buffers.dll
-cp ${SCRIPT_DIR}/../../bin/MessagePack/${BUILDCONFIGURATION}/netstandard2.0/publish/System.Memory.dll ${SCRIPT_DIR}/Assets/Plugins/System.Memory.dll
 cp ${SCRIPT_DIR}/../../bin/MessagePack/${BUILDCONFIGURATION}/netstandard2.0/publish/System.Runtime.CompilerServices.Unsafe.dll ${SCRIPT_DIR}/Assets/Plugins/System.Runtime.CompilerServices.Unsafe.dll
 cp ${SCRIPT_DIR}/../../bin/MessagePack/${BUILDCONFIGURATION}/netstandard2.0/publish/Microsoft.NET.StringTools.dll ${SCRIPT_DIR}/Assets/Plugins/Microsoft.NET.StringTools.dll
-cp ${SCRIPT_DIR}/../../bin/MessagePack/${BUILDCONFIGURATION}/netstandard2.0/publish/System.Threading.Tasks.Extensions.dll ${SCRIPT_DIR}/Assets/Plugins/System.Threading.Tasks.Extensions.dll

